### ISSUE TITLE
test_uct_ib.non_default_gid_idx failure on ThunderX2

### ISSUE THREAD

**Issue by shamisp** [openucx/ucx#3745]
It fails in the same way with rc_mlx5, rc, ud, ud_mlx5:

````
[ RUN      ] rc_mlx5/test_uct_ib.non_default_gid_idx/1
[1561131611.232368] [vulcan2:11593:0]       rc_iface.c:774  UCX  ERROR error modifying QP to RTR: Connection timed out
/ucx.master/contrib/../test/gtest/uct/uct_test.cc:821: Failure
Error: Input/output error
/ucx.master/contrib/../test/gtest/common/test.cc:265: Failure
Failed
Got 1 errors during the test
[1561131611.249089] [vulcan2:11593:0]       rc_iface.c:654  UCX  WARN  some eps were not destroyed
[1561131611.249114] [vulcan2:11593:0]       rc_iface.c:665  UCX  WARN  failed to destroy SRQ: Device or resource busy
[1561131611.249679] [vulcan2:11593:0]       ib_iface.c:911  UCX  WARN  ibv_destroy_cq(recv_cq) returned 16: Device or resource busy
[1561131611.249699] [vulcan2:11593:0]       ib_iface.c:916  UCX  WARN  ibv_destroy_cq(send_cq) returned 16: Device or resource busy
[1561131611.249709] [vulcan2:11593:0]       ib_iface.c:921  UCX  WARN  ibv_destroy_comp_channel(comp_channel) returned 16: Device or resource busy
[  FAILED  ] rc_mlx5/test_uct_ib.non_default_gid_idx/1, where GetParam() = rc_mlx5/mlx5_2:1 (1108 ms)
[----------] 1 test from rc_mlx5/test_uct_ib (1108 ms total)
````

**Comment by yosefe** [comment]
@shamisp @dmitrygladkov is it fixed by #3717?

**Comment by dmitrygladkov** [comment]
> @shamisp @dmitrygladkov is it fixed by #3717?

@yosefe yes, this will be fixed by #3717. I saw the same syndromes w/o the fix on boo31 node
